pkg/cortex/modules.go

```diff
@@ -151,6 +155,10 @@ func (t *Cortex) stopServer() (err error) {
 }
 
 func (t *Cortex) initRing(cfg *Config) (err error) {
+	if cfg.Ingester.LifecyclerConfig.RingConfig.KVStore.Store == kv.MultiClientName {
+		cfg.Ingester.LifecyclerConfig.RingConfig.KVStore.Multi.RuntimeConfigChannel = multiClientRuntimeConfigChannel(t.runtimeConfig)
+	}
+
```

Comment:
I agree that this is ugly. Passing `t.runtimeConfig` to `ring.New` would however introduce cyclic dependency, unless hidden behind some interface. (Also, this runtimeConfig is specific to Cortex, and Loki will need its own). I'll try to implement that.

Reply:
So I've tried to use interface, but that interface would have single function -- to get channel. So now I'm passing that function instead. It's still in the struct though, because MultiClient config should not leak to ring.New API. Take a look and tell me what you think.